Source/WebCore/css/CSSSelector.cpp

```diff
@@ -530,10 +530,8 @@ String CSSSelector::selectorText(StringView separator, StringView rightSide) con
                 break;
 #if ENABLE(FULLSCREEN_API)
             case CSSSelector::PseudoClass::Fullscreen:
-                builder.append(":fullscreen");
-                break;
-            case CSSSelector::PseudoClass::WebKitFullScreen:
-                builder.append(":-webkit-full-screen");
+                // Use serializingValue so alias serializes correctly.
+                builder.append(":"_s, cs->serializingValue());
```

Comment:
Why do we actually want to roundtrip these? It doesn't matter a whole lot, but I kinda like not having to keep around the non-canonical inputs.

Reply:
I think for the purposes of the Web Inspector and CSSOM, it's a better developer experience if `:-webkit-full-screen` stays `:-webkit-full-screen` instead of unexpectedly changing to `:fullscreen`.

Reply:
Is that how CSS aliases generally work? I vaguely recall that they were canonicalized. In DOM we generally canonicalize.

Reply:
Once we decide, I think our decision is best expressed as a test

Reply:
https://drafts.csswg.org/selectors/#legacy-aliasing supports my thinking. https://drafts.csswg.org/css-cascade-5/#legacy-name-alias as well, but for properties only.

I agree that a test would be good. Probably can't be a WPT for this particular one as :-webkit-full-screen isn't standardized, but :matches and :-webkit-autofill are.

Reply:
Looks like you're right: https://github.com/w3c/csswg-drafts/issues/5847
I'll add an internal test, `:-webkit-autofill` already has a test (although it allows serializing to either selectors).

I wonder how well this plays out with WebInspector though, I guess it's not a big deal for CSSOM, but having WebInspector show a selector different than what was authored might throw off a little bit. Perhaps the WebInspector doesn't use CSSOM for this case though.

Reply:
WebInspector doesn't appear to use CSSOM for this case, all good here